zhaquirks/bosch/rbsh_trv0_zb_eu.py

```diff
@@ -0,0 +1,589 @@
+            update_attributes = [
+                attr
+                for attr in args.attribute_reports
+                if attr.attrid != SYSTEM_MODE_ATTR.id
+            ]
+            if len(update_attributes) > 0:
+                msg = foundation.GENERAL_COMMANDS[
+                    foundation.GeneralCommand.Report_Attributes
+                ].schema(attribute_reports=update_attributes)
+                return super().handle_cluster_general_request(
+                    hdr, msg, dst_addressing=dst_addressing
+                )
+
+
+class BoschUserInterfaceCluster(CustomCluster, UserInterface):
+    """Bosch UserInterface cluster."""
+
+    class AttributeDefs(UserInterface.AttributeDefs):
+        """Bosch user interface manufacturer specific attributes."""
+
+        display_orientation: Final = ZCLAttributeDef(
+            id=SCREEN_ORIENTATION_ATTR_ID,
+            # To be matched to BoschDisplayOrientation enum.
+            type=t.uint8_t,
+            is_manufacturer_specific=True,
+        )
+
+        display_on_time: Final = ZCLAttributeDef(
+            id=SCREEN_TIMEOUT_ATTR_ID,
+            # Usable values range from 5-30
+            type=t.enum8,
+            is_manufacturer_specific=True,
+        )
+
+        display_brightness: Final = ZCLAttributeDef(
+            id=SCREEN_BRIGHTNESS_ATTR_ID,
+            # Values range from 0-10
+            type=t.enum8,
+            is_manufacturer_specific=True,
+        )
+
+        displayed_temperature: Final = ZCLAttributeDef(
+            id=DISPLAY_MODE_ATTR_ID,
+            type=BoschDisplayedTemperature,
+            is_manufacturer_specific=True,
+        )
+
+    async def write_attributes(
+        self, attributes: dict[str | int, Any], manufacturer: int | None = None
+    ) -> list:
+        """display_orientation special handling.
+
+        - convert from enum to uint8_t
+        """
+        display_orientation_attr = self.AttributeDefs.display_orientation
+
+        remaining_attributes = attributes.copy()
+        display_orientation_attribute_id = None
+
+        """Check if display_orientation is being written (can be numeric or string)."""
+        if display_orientation_attr.id in attributes:
+            display_orientation_attribute_id = display_orientation_attr.id
+        elif display_orientation_attr.name in attributes:
+            display_orientation_attribute_id = display_orientation_attr.name
+
+        if display_orientation_attribute_id is not None:
+            display_orientation_value = remaining_attributes.pop(
+                display_orientation_attribute_id
+            )
+            new_display_orientation_value = DISPLAY_ORIENTATION_ENUM_TO_INT_MAP[
+                display_orientation_value
+            ]
+            remaining_attributes[display_orientation_attribute_id] = (
+                new_display_orientation_value
+            )
+
+        return await super().write_attributes(remaining_attributes, manufacturer)
```

Comment:
To avoid this, we could either just introduce a switch "Flip display" or we might be able to use `zcl_type` on the `ZCLAttributeDef` like this:
https://github.com/zigpy/zha-device-handlers/blob/eda673936ae01ea50fc7806e3fc8ae57fa3432a1/zhaquirks/nodon/pilot_wire.py#L42

Just treat everything as if it's an enum attribute and add that. That might fix it, so we can remove this block (and also from tests).